### PR TITLE
HTTP client treat non 20X status codes

### PR DESCRIPTION
**Pull request by gilmarsquinelato**
HTTP client now throws an exception according to the response status code if >= 400

### CODE REVIEW

File: src/errors/index.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 export * from './transport';
 export * from './base';
-export * from './not_found';
-export * from './forbidden';
-export * from './unauthorized';
-export * from './internal';
+export * from './notFoundError';
+export * from './forbiddenError';
+export * from './unauthorizedError';
+export * from './internalServerError';
```
Comment (fenos) [inline src/errors/index.ts]: we use `.error` convention in the file name, ex: `forbidden.error.ts`

File: src/http/http.errors.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const httpErrorMapping: Record<number, typeof BaseError> = {
+  400: BadRequestError,
+  401: UnauthorizedError,
+  403: ForbiddenError,
+  404: NotFoundError,
+  500: InternalServerError,
+};
+
+const getErrorFromStatusCode = (statusCode: number): typeof BaseError =>
```
Comment (fenos) [inline src/http/http.errors.ts]: You don't have to change it. 
But you can make this function use inferred types from the argument as following:

[TS-Playground](http://www.typescriptlang.org/play/#code/MYGwhgzhAEBCkFMCiAnFB7F0DeBfAUKJDPACYBKCAjgK4IQAuqGW2+00AdtALzQCM+AkSjQAqpzA0GAC0wBLAF4JSzTDnZde0AExDC4UQDFMAI3mlSCTmtabufAMz6RMAHLoGJmp1Vp1bBwO0AAsLoYwAJKcDAgokiAAynEAbnG2Ghyk2vwA3PbaAKz6+AgAHgAOmAzQwOicjNAyDAwVtgCyYBUV8pwA5tqBoQAMwwBccGAU1HSMtgA0miHD-BMSUrIKyn4sixzLjhMmKOaW1gtLwyETHl7oPjuYe9CFoxPRsfFgSanp-iiLXD5Qj1Rp9BBMf5GDAAW0SDDADBoEAAwugrNoADwAFWg5VivhgAGsEABPdAAM2gAAoGKSKghKU0Wm1-p1ur0+gBKAB81MYiORaKsE1xAB8uDQYaY4lyJrT6Yyqc1Wh0uj1+lyANrYgC6vB5mgFSNR6IQ0F6zNVbPVnOgAH4rayWOyNX0tcahWboJBoHroBMVc7MK7OVrXsNdcCQQ0aiBtODISxoeg4QiTcKENTliEuflOAgAO7QEDUrkAOk4+Gr5SqKBqVgpUhANUTthTacFpqsuSAA)

File: src/http/http.client.ts [hunk 3]
```diff
@@ -52,15 +51,23 @@ class Client {
...
       timeRequest();
       throw new TransportError(error.message);
-    });
+    }
+
+    labels.status = response.status;
+    timeRequest();
+
+    if (response.status >= 400) {
+      throw new (getErrorFromStatusCode(response.status) as any);
```
Comment (fenos) [inline src/http/http.client.ts]: Why as any?

File: src/errors/index.ts (hunk 1 above)
Comment (gilmarsquinelato) [reply src/errors/index.ts]: renamed the files

File: src/http/http.client.ts (hunk 3 above)
Comment (gilmarsquinelato) [reply src/http/http.client.ts]: because the `new` wasn't allowed to instantiate the return of a function :(

File: src/http/http.errors.ts (hunk 2 above)
Comment (gilmarsquinelato) [reply src/http/http.errors.ts]: removed the explicit return types rule from ESLint
changed to infer the return type

File: src/http/http.client.ts (hunk 3 above)
Comment (fenos) [reply src/http/http.client.ts]: is there a way to create an instance of the `BaseError` without using `any`?

**Pull request merged**